v2/oauth/multiple-frontends-with-separate-backends.mdx

```diff
@@ -0,0 +1,427 @@
+        return
+    }
+    defer response.Body.Close()
+
+    var user User
+    if err := json.NewDecoder(response.Body).Decode(&user); err != nil {
+        http.Error(w, "Failed to decode user info", http.StatusInternalServerError)
+        return
+    }
+
+    session.Values["user"] = user
+    session.Save(r, w)
+
+    http.Redirect(w, r, "/", http.StatusFound)
+}
+
+func logoutHandler(w http.ResponseWriter, r *http.Request) {
+    session, _ := store.Get(r, "session")
+    delete(session.Values, "user")
+    session.Save(r, w)
+
+    http.Redirect(w, r, "/login", http.StatusFound)
+}
+```
+
+
+</TabItem>
+
+
+<TabItem value="python"> 
+
+```python
+from flask import Flask, redirect, url_for, session, request
+from authlib.integrations.flask_client import OAuth
+from flask_session import Session
+
+# Configuration
+AUTHORIZATION_SERVER_URL = 'https://auth.example.com'
+# The client ID returned from the first step
+CLIENT_ID = 'your-client-id'
+# The client secret returned from the first step
+CLIENT_SECRET = 'your-client-secret'
+# One of the callback URLs specified in the create client payload 
+CALLBACK_URL = 'https://your-app.com/auth/callback' 
+
+app = Flask(__name__)
+app.secret_key = 'your-session-secret'
+
+# Configure server-side session management
+app.config['SESSION_TYPE'] = 'filesystem'
+app.config['SESSION_PERMANENT'] = False
+Session(app)
+
+oauth = OAuth(app)
+oauth.register(
+    name='auth_provider',
+    client_id=CLIENT_ID,
+    client_secret=CLIENT_SECRET,
+    authorize_url=f'{AUTHORIZATION_SERVER_URL}/authorize',
+    authorize_params=None,
+    access_token_url=f'{AUTHORIZATION_SERVER_URL}/token',
+    access_token_params=None,
+    refresh_token_url=None,
+    redirect_uri=CALLBACK_URL,
+    client_kwargs={'scope': 'openid profile email'},
+)
+
+@app.route('/')
+def index():
+    user = session.get('user')
+    if user:
+        return f'Hello, {user["name"]}'
+    return 'You are not logged in'
+
+@app.route('/login')
+def login():
+    redirect_uri = url_for('auth_callback', _external=True)
+    return oauth.auth_provider.authorize_redirect(redirect_uri)
+
+@app.route('/auth/callback')
+def auth_callback():
+    token = oauth.auth_provider.authorize_access_token()
+    user = oauth.auth_provider.parse_id_token(token)
+    session['user'] = user
+    return redirect('/')
+
+@app.route('/logout')
+def logout():
+    session.pop('user', None)
+    return redirect('/')
+
+@app.errorhandler(401)
+def unauthorized(e):
+    return 'Unauthorized'
+
+@app.errorhandler(403)
+def forbidden(e):
+    return 'Forbidden'
+
+@app.errorhandler(404)
+def page_not_found(e):
+    return 'Page not found'
+
+# Start the server
+if __name__ == '__main__':
+    app.run(debug=True)
+```
+
+</TabItem>
+
+
+</BackendSDKTabs>
+
+
+## 4. Update the login flow in your frontend applications
+
+In your `frontend applications` you will have to add a login action that will direct the user to the authentication page.
```

Comment:
I dont think this is all thats needed either. What about things like state generation and verification etc? What about mobile apps or desktop apps? We need to rethink this guide from step 3 onwards.

Reply:
Hmm, we could add guidance on those but in general I think we should recommend using libraries that take care of it. I can check how this works in `passport`.

Reply:
What do you mean by state generation and verification?